packages/wrangler/src/pages/dev.tsx

```diff
@@ -307,8 +310,44 @@ export const Handler = async ({
 		);
 	}
 
+	let entrypoint = scriptPath;
+
+	if (directory) {
+		const routesJSONPath = join(directory, "_routes.json");
+
+		if (existsSync(routesJSONPath)) {
+			const routesJSONContents = readFileSync(routesJSONPath, "utf-8");
+
+			try {
+				validateRoutes(JSON.parse(routesJSONContents), directory);
+			} catch (err) {
+				throw err;
+			}
+
+			entrypoint = join(tmpdir(), `${Math.random().toString(36).slice(2)}.js`);
```

Comment:
I know I wrote this bit, and it's 100% not required for this PR, but this is becoming a frequently used enough pattern, that I'd be tempted to make a function out of this: `makeTemporaryRandomFile(ext: string)`

Reply:
will make a separate PR out of that, mainly because most of the places we generate these temp random files, we usually specify the file name and randomise just the (let's call it) 'hash' segment. Plus we'd need to support file extensions as well, since we'd be calling this to generate json/esm/js files (from scanning our pages code). So I feel this should have a scope of its own. Sound good?

Reply:
Yup, ofc. Like I said, totally not required in this PR. And honestly, not even something you need to do. I flagged it mostly for myself.